### PR TITLE
Programming Languages - "Haskell" Complete Resources

### LINKED ISSUE #1006: 💡 [FEATURE] -  Programming Languages - "Haskell" Complete Resources
### Idea Contribution

- [X] I have read all the feature request issues.
- [X] I'm interested in working on this issue
- [X] I'm part of GSSOC organization

### Explain feature request

Adding Programming Language - "Haskell" Complete Resources

### PR DESCRIPTION
**Pull request by kosuri-indu**
## Description

Hii @jfmartinz
I have added Haskell resources in programming languages directory.
image

Fixes #1006

## Type of change
- [x] Adding a new Resource
- [x] Adding a new Category
- [x] Documentation enhancement or fixes

## Checklist:
- [x] I read carefully [CONTRIBUTING.md](https://github.com/jfmartinz/ResourceHub/blob/main/CONTRIBUTING.md)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by kosuri-indu** [comment]
@jfmartinz 
Please review this PR :)
Thank you

File: Programming Languages/Haskell/readme.md [hunk 1]
```diff
@@ -0,0 +1,560 @@
...
+  </tr>
+</table>
+
+## Tools
+
+> Essential tools and libraries for Haskell development.
+
+<table width="100%" id="tools">
+  <tr>
+    <th>Tool</th>
```
Comment (jfmartinz) [inline Programming Languages/Haskell/readme.md:343]:
```suggestion
    <th>Resource Name</th>
```

File: Programming Languages/Haskell/readme.md [hunk 2]
```diff
@@ -0,0 +1,560 @@
...
+
+## Projects
+
+> Hands-on projects to apply and reinforce your Haskell skills.
+
+### Beginner Level Projects
+
+<table width="100%" id="beginner-projects">
+  <tr>
+    <th>Project</th>
```
Comment (jfmartinz) [inline Programming Languages/Haskell/readme.md:372]:
```suggestion
    <th>Resource Name</th>
```

File: Programming Languages/Haskell/readme.md [hunk 3]
```diff
@@ -0,0 +1,560 @@
...
+    <td><a href="https://www.youtube.com/watch?v=5URsWAlkT_0&list=PLMTONe7-tohlyDmVN7mpkqJZNUmRlW6rg">Make you a roguelike in Haskell</a></td>
+    <td>Create a basic roguelike game in Haskell, learning about game development and functional programming.</td>
+  </tr>
+</table>
+
+### Intermediate Level Projects
+
+<table width="100%" id="intermediate-projects">
+  <tr>
+    <th>Project</th>
```
Comment (jfmartinz) [inline Programming Languages/Haskell/readme.md:393]:
```suggestion
    <th>Resource Name</th>
```

File: Programming Languages/Haskell/readme.md [hunk 4]
```diff
@@ -0,0 +1,560 @@
...
+    <td><a href="https://www.youtube.com/watch?v=N9RUqGYuGfw&list=PLMTONe7-tohlyDmVN7mpkqJZNUmRlW6rg&index=4">Json Parser from Scratch</a></td>
+    <td>Learn to build a JSON parser from scratch in Haskell, focusing on parsing techniques and data manipulation.</td>
+  </tr>
+</table>
+
+### Advanced Level Projects
+
+<table width="100%" id="advanced-projects">
+  <tr>
+    <th>Project</th>
```
Comment (jfmartinz) [inline Programming Languages/Haskell/readme.md:410]:
```suggestion
    <th>Resource Name</th>
```

File: Programming Languages/Haskell/readme.md [hunk 5]
```diff
@@ -0,0 +1,560 @@
...
+
+## Courses
+
+> Recommended courses for learning Haskell.
+
+### Free Courses
+
+<table width="100%" id="free-courses">
+  <tr>
+    <th>Course</th>
```
Comment (jfmartinz) [inline Programming Languages/Haskell/readme.md:435]:
```suggestion
    <th>Resource Name</th>
```

File: Programming Languages/Haskell/readme.md [hunk 6]
```diff
@@ -0,0 +1,560 @@
...
+    <td><a href="https://exercism.org/tracks/haskell">Exercism Haskell Track</a></td>
+    <td>Practice Haskell programming with a series of exercises on Exercism's Haskell track.</td>
+  </tr>
+</table>
+
+### Paid Courses
+
+<table width="100%" id="paid-courses">
+  <tr>
+    <th>Course</th>
```
Comment (jfmartinz) [inline Programming Languages/Haskell/readme.md:468]:
```suggestion
    <th>Resource Name</th>
```

File: Programming Languages/Haskell/readme.md [hunk 7]
```diff
@@ -0,0 +1,560 @@
...
+  </tr>
+</table>
+
+## Youtube
+
+> Video tutorials and lectures for learning Haskell.
+
+<table width="100%" id="youtube-videos">
+  <tr>
+    <th>Video</th>
```
Comment (jfmartinz) [inline Programming Languages/Haskell/readme.md:495]:
```suggestion
    <th>Resource Name</th>
```

File: Programming Languages/Haskell/readme.md [hunk 8]
```diff
@@ -0,0 +1,560 @@
...
+  </tr>
+</table>
+
+## Haskell Cheatsheets
+
+> Quick references to Haskell syntax and concepts.
+
+<table width="100%" id="haskell-cheatsheets">
+  <tr>
+    <th>Cheatsheet</th>
```
Comment (jfmartinz) [inline Programming Languages/Haskell/readme.md:522]:
```suggestion
    <th>Resource Name</th>
```

**Review by jfmartinz — CHANGES_REQUESTED** [review]

**Comment by jfmartinz** [comment]
Give me the issue number please

**Comment by kosuri-indu** [comment]
> Give me the issue number please

Fixes #1006

**Review by jfmartinz — APPROVED** [review]

**Pull request merged**